controllers/util.go

```diff
@@ -86,22 +94,75 @@ func (r *L2ServiceAttachmentReconciler) DefineNetAttachDef(ctx context.Context,
 	// Initiate L2ServiceAttachment Parser
 	l2srvAttParser := l2serviceattachmentparser.NewL2SrvAttParser(s, l2srvObjs, cp, subnets, routesMap, r.Config, r.CniMap, r.IpamMap, log)
 	// Parse the resources and fill the data
-	manifestFolder, err := l2srvAttParser.ParseL2ServiceAttachment(&data)
+	cniManifestFile, ipamManifestFile, err := l2srvAttParser.ParseL2ServiceAttachment(data)
 	if err != nil {
 		log.Error(err, "Error occurred during parsing the L2ServiceAttachment")
 		return nil, err
 	}
 
-	data.Data["NetAttachDefName"] = s.Name
-	data.Data["NetAttachDefNamespace"] = s.Namespace
+	data["NetAttachDefName"] = s.Name
+	data["NetAttachDefNamespace"] = s.Namespace
+	if prefix, ok := data["ResourcePrefix"]; ok {
```

Comment:
Where do we set this key ? data["ResourcePrefix"]

Reply:
I think it got missed out during the merge, added this in cni.go file